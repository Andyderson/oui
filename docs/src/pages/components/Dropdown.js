
  import React from 'react';
  import ComponentContent from '../../layouts/templates/ComponentContent';

   
 import examples from '../../../../src/components/Dropdown/example'; import readme from 'raw-loader!../../../../src/components/Dropdown/README.md';

  const sassData = {"name":"Dropdown\n","description":"","start":{"line":1,"column":1},"end":{"line":92,"column":4},"example":[{"type":"html","description":"Dropdown default","code":"<div style=\"min-height: 150px;\">\n  <div class=\"#{$namespace}dropdown-group\">\n    <button class=\"oui-button\">\n      Dropdown <span class=\"oui-arrow-inline--down\"></span>\n    </button>\n    <ul class=\"#{$namespace}dropdown width--200\" style=\"display: block;\">\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Manage Collaborators</a>\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Manage This Thing</a>\n      </li>\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Experiment Change History</a>\n      </li>\n    </ul>\n  </div>\n</div>"},{"type":"html","description":"Dropdown right","code":"<div class=\"text--right\" style=\"min-height: 150px;\">\n  <div class=\"#{$namespace}dropdown-group\">\n    <button class=\"oui-button\">\n      Dropdown <span class=\"oui-arrow-inline--down\"></span>\n    </button>\n    <ul class=\"#{$namespace}dropdown #{$namespace}dropdown--right width--200\" style=\"display: block;\">\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Manage Collaborators</a>\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Manage This Thing</a>\n      </li>\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Experiment Change History</a>\n      </li>\n    </ul>\n  </div>\n</div>"},{"type":"html","description":"Dropdown with filter","code":"<div style=\"min-height: 150px;\">\n  <div class=\"#{$namespace}dropdown-group\">\n    <button class=\"oui-button\">\n      Dropdown <span class=\"oui-arrow-inline--down\"></span>\n    </button>\n    <ul class=\"#{$namespace}dropdown width--200\" style=\"display: block;\">\n      <li class=\"#{$namespace}dropdown__item\">\n        <form class=\"soft-half--ends\">\n          <input type=\"text\" class=\"oui-text-input oui-text-input--search\">\n        </form>\n      </li>\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Experiment Change History</a>\n      </li>\n    </ul>\n  </div>\n</div>"},{"type":"html","description":"Dropdown inside input","code":"<div style=\"min-height: 150px;\">\n  <div class=\"#{$namespace}dropdown-group\">\n    <input type=\"text\" class=\"oui-text-input\" placeholder=\"Input with a Dropdown\">\n    <ul class=\"#{$namespace}dropdown width--200\" style=\"display: block;\">\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Manage Collaborators</a>\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Manage This Thing</a>\n      </li>\n      <li class=\"#{$namespace}dropdown__item\">\n        <a href=\"#\" class=\"#{$namespace}dropdown__block-link\">Experiment Change History</a>\n      </li>\n    </ul>\n  </div>\n</div>"},{"type":"html","description":"Descriptive dropdown","code":"<div style=\"min-height: 190px;\">\n  <div class=\"#{$namespace}dropdown-group\">\n    <button class=\"oui-button\">\n      Dropdown <span class=\"arrow-inline--down\"></span>\n    </button>\n    <ul class=\"#{$namespace}dropdown dropdown--descriptive width--250\" style=\"display: block;\">\n      <li class=\"#{$namespace}dropdown__item\">\n        <div class=\"#{$namespace}dropdown__block-link\">\n          <div class=\"#{$namespace}dropdown--descriptive__header\">Improvement over time</div>\n          <div class=\"#{$namespace}dropdown--descriptive__content\">Cumulative improvement over time for the personalized audience compared to the holdback.</div>\n        </div>\n      </li>\n      <li class=\"#{$namespace}dropdown__item\">\n        <div class=\"#{$namespace}dropdown__block-link\">\n          <div class=\"#{$namespace}dropdown--descriptive__header\">Conversions over time</div>\n          <div class=\"#{$namespace}dropdown--descriptive__content\">Daily conversions over time for both the personalized and holdback audiences.</div>\n        </div>\n      </li>\n    </ul>\n  </div>\n</div>"}]};
  const reactData = {"description":"","displayName":"Dropdown","methods":[{"name":"handleMouseOverChildren","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"handleMouseLeavingChildren","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"handleOnBlur","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"handleToggle","docblock":null,"modifiers":[],"params":[{"name":"event","type":null}],"returns":null}],"props":{"buttonContent":{"type":{"name":"union","value":[{"name":"string"},{"name":"element"}]},"required":false,"description":"Button text, can be a string or element."},"children":{"type":{"name":"node"},"required":true,"description":"Dropdown contents, typically using the <Blocklist> component."},"fullWidth":{"type":{"name":"bool"},"required":false,"description":"Button width is either full or inline-block."},"handleClick":{"type":{"name":"func"},"required":false,"description":"Unused..."},"icon":{"type":{"name":"bool"},"required":false,"description":"Show a dropdown arrow, true/false."},"isDisabled":{"type":{"name":"bool"},"required":false,"description":"Disable button."},"style":{"type":{"name":"string"},"required":false,"description":"Button style, e.g. highlight, danger, outline."},"testSection":{"type":{"name":"string"},"required":false,"description":"For automated testing only."},"width":{"type":{"name":"union","value":[{"name":"string"},{"name":"number"}]},"required":false,"description":"Dropdown menu width, in pixels."},"zIndex":{"type":{"name":"number"},"required":false,"description":"Override default dropdown menu z-index."}},"private":false};

  let dataObject = {
    react: reactData, 
    examples: examples ? examples : null,
    readme: readme ? readme : null,
    sass: sassData,
  }
  
  class DropdownComponent extends React.Component {
  
    render() {
      return (
        <ComponentContent data={ dataObject } />
      );
    };
  }
  
  export default DropdownComponent
