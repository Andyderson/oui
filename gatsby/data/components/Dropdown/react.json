{"description":"","displayName":"Dropdown","methods":[{"name":"handleMouseOverChildren","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"handleMouseLeavingChildren","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"handleOnBlur","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"handleToggle","docblock":null,"modifiers":[],"params":[{"name":"event","type":null}],"returns":null}],"props":{"buttonContent":{"type":{"name":"union","value":[{"name":"string"},{"name":"element"}]},"required":false,"description":"Button text, can be a string or element."},"children":{"type":{"name":"node"},"required":true,"description":"Dropdown contents, typically using the <Blocklist> component."},"fullWidth":{"type":{"name":"bool"},"required":false,"description":"Button width is either full or inline-block."},"handleClick":{"type":{"name":"func"},"required":false,"description":"Unused..."},"icon":{"type":{"name":"bool"},"required":false,"description":"Show a dropdown arrow, true/false."},"isDisabled":{"type":{"name":"bool"},"required":false,"description":"Disable button."},"style":{"type":{"name":"string"},"required":false,"description":"Button style, e.g. highlight, danger, outline."},"testSection":{"type":{"name":"string"},"required":false,"description":"For automated testing only."},"width":{"type":{"name":"union","value":[{"name":"string"},{"name":"number"}]},"required":false,"description":"Dropdown menu width, in pixels."},"zIndex":{"type":{"name":"number"},"required":false,"description":"Override default dropdown menu z-index."}},"private":false}