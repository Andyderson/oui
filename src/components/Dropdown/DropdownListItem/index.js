import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export default function DropdownListItem(props) {
  function onClick() {
    if (props.hideOnClick) {
      props.handleToggle();
    }
  }

  const classes = classNames({
    'hard--sides': props.hardSides,
    'hard--top': props.hardTop,
    'oui-dropdown__item': true,
  });

  return (
    <li
      className={ classes }
      onClick={ onClick }>
      { props.children }
    </li>
  );
}

DropdownListItem.propTypes = {
  /** Nodes to display within */
  children: PropTypes.node,
  /** Function to toggle visibility of Dropdown.Contents */
  handleToggle: PropTypes.func,
  /** Remove padding from sides */
  hardSides: PropTypes.bool,
  /** Remove padding from top */
  hardTop: PropTypes.bool,
  /** Whether to hide Dropdown.Contents when the item is clicked or not */
  hideOnClick: PropTypes.bool,
};

DropdownListItem.defaultProps = {
  hideOnClick: true,
};
