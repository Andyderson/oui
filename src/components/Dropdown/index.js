import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Manager, Target, Popper } from 'react-popper';

class Dropdown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      overChildren: false,
    };
    this.handleToggle = this.handleToggle.bind(this);
    this.handleOnBlur = this.handleOnBlur.bind(this);
    this.handleMouseOverChildren = this.handleMouseOverChildren.bind(this);
    this.handleMouseLeavingChildren = this.handleMouseLeavingChildren.bind(this);
  }

  handleMouseOverChildren() {
    this.setState({
      overChildren: true,
    });
  }

  handleMouseLeavingChildren() {
    this.setState({
      overChildren: false,
    });
  }

  handleOnBlur() {
    if (!this.state.overChildren) {
      this.setState({
        isOpen: false,
      });
    }
  }

  handleToggle(event) {
    this.setState({
      isOpen: !this.state.isOpen,
      overChildren: false,
    });
  }

  render() {
    const {
      buttonContent,
      children,
      icon = true,
      isDisabled = false,
      fullWidth,
      style,
      testSection,
      width = 200,
      zIndex = 999,
    } = this.props;

    const groupClass = classNames(
      'oui-dropdown-group',
      'target',
      { ['width--1-1']: fullWidth }
    );

    const buttonClass = classNames(
      'oui-button',
      { [`oui-button--${style}`]: style },
      { ['oui-button--full soft--left text--left']: fullWidth }
    );

    const iconClass = classNames(
      'push-half--left',
      {
        ['oui-arrow-inline--down']: icon,
      }
    );

    return (
      <Manager>
        <Target data-ui-component={ true }
          className={ groupClass }
          data-test-section={ testSection }>
          <button
            type='button'
            className={ buttonClass }
            disabled={ isDisabled }
            onClick={ this.handleToggle }
            onBlur={ this.handleOnBlur }>
            <div className='flex'>
              <div className='flex--1 truncate'>{ buttonContent }</div>
              <div className='text--right'><span className={ iconClass }></span></div>
            </div>
          </button>
        </Target>
        <Popper placement='bottom-start' className='popper'>
          <div
            className='oui-dropdown-children'
            style={{zIndex: zIndex, position: 'absolute', width: width}}
            onMouseOver={ this.handleMouseOverChildren }
            onMouseLeave={ this.handleMouseLeavingChildren }
            onClick={ this.handleToggle } >
            { this.state.isOpen && !isDisabled && children }
          </div>
        </Popper>
      </Manager>
    );
  }
}

Dropdown.propTypes = {
  /** Button text, can be a string or element. */
  buttonContent: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
  ]),
  /** Dropdown contents, typically using the <Blocklist> component. */
  children: PropTypes.node.isRequired,
  /** Button width is either full or inline-block. */
  fullWidth: PropTypes.bool,
  /** Unused... */
  handleClick: PropTypes.func,
  /** Show a dropdown arrow, true/false. */
  icon: PropTypes.bool,
  /** Disable button. */
  isDisabled: PropTypes.bool,
  /** Button style, e.g. highlight, danger, outline. */
  style: PropTypes.string,
  /** For automated testing only. */
  testSection: PropTypes.string,
  /** Dropdown menu width, in pixels. */
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  /** Override default dropdown menu z-index. */
  zIndex: PropTypes.number,
};

export default Dropdown;
