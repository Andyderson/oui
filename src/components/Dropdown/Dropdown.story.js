import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, number, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Dropdown from './index.js';
import BlockList from '../BlockList';
import Icon from 'react-oui-icons';

const data = [
  {title: 'Manual', description: 'Dolcelatte cheeseburger swiss paneer cow gouda edam cheese slices'},
  {title: 'Maximize Conventions', header: 'Automation',
    description: 'Everyone loves cut the cheese cow squirty cheese emmental jarlsberg halloumi camembert de normandie.'},
  {title: 'Faster Results', description: 'Cauliflower cheese stilton paneer hard cheese fondue.'},
];

const stories = storiesOf('Dropdown', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.addWithInfo('with BlockList elements', () => {
  return (
    <Dropdown
      isDisabled={ boolean('isDisabled', false) }
      fullWidth={ boolean('fullWidth', false) }
      style={ select('style', {
        default: 'default',
        highlight: 'highlight',
        danger: 'danger',
        'danger-outline': 'danger-outline',
        outline: 'outline',
        underline: 'underline'}, '') }
      buttonContent='Dropdown'
      width={ number('width', 300) }>
      <BlockList>
        {
          data.map((item, index) => {
            return (
              <BlockList.Category header={ item.header } key={ index }>
                <BlockList.Item onClick={ action('click on complex item') }>
                  <div className="flex flex-align--center">
                    <div className="flex--1">
                      <div>{ item.title }</div>
                      <div className="muted micro">{ item.description }</div>
                    </div>
                  </div>
                </BlockList.Item>
              </BlockList.Category>
            );
          })
        }
      </BlockList>
    </Dropdown>
  );
});


stories.addWithInfo('three dots', () => {
  return (
    <Dropdown
      icon={ false }
      isDisabled={ boolean('isDisabled', false) }
      fullWidth={ boolean('fullWidth', false) }
      buttonContent={ <Icon name='ellipsis' /> }
      width={ number('width', 100) }>
      <BlockList>
        <BlockList.Category >
          <BlockList.Item onClick={ action('click on complex item 2') }>
            <div className="flex flex-align--center">
              <div className="flex--1">
                <a href='#'>un link </a>
              </div>
            </div>
          </BlockList.Item>
        </BlockList.Category>
        <BlockList.Category >
          <BlockList.Item onClick={ action('click on complex item 3') }>
            <div className="flex flex-align--center">
              <div className="flex--1">
                <a href='#'>otro link</a>
              </div>
            </div>
          </BlockList.Item>
        </BlockList.Category>
        <BlockList.Category >
          <BlockList.Item onClick={ action('click on complex item 4') }>
            <div className="flex flex-align--center">
              <div className="flex--1">
                <a href='#'>un más</a>
              </div>
            </div>
          </BlockList.Item>
        </BlockList.Category>
        <BlockList.Category >
          <BlockList.Item onClick={ action('click on complex item 5') }>
            <div className="flex flex-align--center">
              <div className="flex--1">
                <a href='#'>el último</a>
              </div>
            </div>
          </BlockList.Item>
        </BlockList.Category>
      </BlockList>
    </Dropdown>
  );
});


stories.addWithInfo('text behind', () => {
  return (
    <div>
      <Dropdown
        isDisabled={ boolean('isDisabled', false) }
        fullWidth={ boolean('fullWidth', false) }

        buttonContent='Dropdown'
        width={ number('width', 100) }>
        <BlockList>
          <BlockList.Category >
            <BlockList.Item onClick={ action('click on complex item') }>
              <div className="flex flex-align--center">
                <div className="flex--1">
                  <a href='#'>un link </a>
                </div>
              </div>
            </BlockList.Item>
          </BlockList.Category>
          <BlockList.Category >
            <BlockList.Item onClick={ action('click on complex item') }>
              <div className="flex flex-align--center">
                <div className="flex--1">
                  <a href='#'>otro link</a>
                </div>
              </div>
            </BlockList.Item>
          </BlockList.Category>
          <BlockList.Category >
            <BlockList.Item onClick={ action('click on complex item') }>
              <div className="flex flex-align--center">
                <div className="flex--1">
                  <a href='#'>un más</a>
                </div>
              </div>
            </BlockList.Item>
          </BlockList.Category>
          <BlockList.Category >
            <BlockList.Item onClick={ action('click on complex item') }>
              <div className="flex flex-align--center">
                <div className="flex--1">
                  <a href='#'>el último</a>
                </div>
              </div>
            </BlockList.Item>
          </BlockList.Category>
        </BlockList>
      </Dropdown>
      <h1>This text should be behind the open dropdown</h1>
    </div>
  );
});
