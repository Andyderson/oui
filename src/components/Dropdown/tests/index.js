import React from 'react';
import Dropdown from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

const data = [
  {title: 'Manual', description: 'Dolcelatte cheeseburger swiss paneer cow gouda edam cheese slices'},
  {title: 'Maximize Conventions', header: 'Automation',
    description: 'Everyone loves cut the cheese cow squirty cheese emmental jarlsberg halloumi camembert de normandie.'},
  {title: 'Faster Results', description: 'Cauliflower cheese stilton paneer hard cheese fondue.'},
];

describe('components/Dropdown', () => {
  it('should not render children when isOpen is false', () => {
    const output = shallow(
      <Dropdown
        icon='chevron'
        text='Dropdown'
        isOpen={ false }>
        <ul>
          { data.map((item, index) => {
            return (
              <li key={ index }>{ item.title }</li>
            );
          })
          }
        </ul>
      </Dropdown>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render children when isOpen is true', () => {
    const output = shallow(
      <Dropdown
        icon='chevron'
        text='Dropdown'
        isOpen={ true }>
        <ul>
          { data.map((item, index) => {
            return (
              <li key={ index }>{ item.title }</li>
            );
          })
          }
        </ul>
      </Dropdown>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should not render children when isDisabled is true', () => {
    const output = shallow(
      <Dropdown
        icon='chevron'
        text='Dropdown'
        idDisabled={ true }>
        <ul>
          { data.map((item, index) => {
            return (
              <li key={ index }>{ item.title }</li>
            );
          })
          }
        </ul>
      </Dropdown>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should not use .oui-arrow-inline--down when icon isEqual to triangle', () => {
    const output = shallow(
      <Dropdown
        icon='triangle'
        text='Dropdown'>
        <ul>
          { data.map((item, index) => {
            return (
              <li key={ index }>{ item.title }</li>
            );
          })
          }
        </ul>
      </Dropdown>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should use oui-button--full class when fullWidth is true', () => {
    const output = shallow(
      <Dropdown
        icon='triangle'
        text='Dropdown'
        fullWidth={ true }>
        <ul>
          { data.map((item, index) => {
            return (
              <li key={ index }>{ item.title }</li>
            );
          })
          }
        </ul>
      </Dropdown>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should highlight button class when style equals highlight', () => {
    const output = shallow(
      <Dropdown
        icon='triangle'
        text='Dropdown'
        style='highlight'>
        <ul>
          { data.map((item, index) => {
            return (
              <li key={ index }>{ item.title }</li>
            );
          })
          }
        </ul>
      </Dropdown>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
