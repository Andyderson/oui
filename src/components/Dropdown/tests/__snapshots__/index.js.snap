// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropdown should highlight button class when style equals highlight 1`] = `
<div
  className="lego-dropdown-group"
  data-ui-component={true}
  style={
    Object {
      "width": 200,
    }
  }
>
  <button
    className="oui-button oui-button--highlight"
    disabled={false}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="flex--1 truncate"
      >
        Dropdown
      </div>
      <div
        className="text--right"
      >
        <span
          className="push-half--left oui-arrow-inline--down"
        />
      </div>
    </div>
  </button>
</div>
`;

exports[`components/Dropdown should not render children when isDisabled is true 1`] = `
<div
  className="lego-dropdown-group"
  data-ui-component={true}
  style={
    Object {
      "width": 200,
    }
  }
>
  <button
    className="oui-button"
    disabled={false}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="flex--1 truncate"
      >
        Dropdown
      </div>
      <div
        className="text--right"
      >
        <span
          className="push-half--left lego-arrow-inline--down"
        />
      </div>
    </div>
  </button>
</div>
`;

exports[`components/Dropdown should not render children when isOpen is false 1`] = `
<div
  className="lego-dropdown-group"
  data-ui-component={true}
  style={
    Object {
      "width": 200,
    }
  }
>
  <button
    className="oui-button"
    disabled={false}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="flex--1 truncate"
      >
        Dropdown
      </div>
      <div
        className="text--right"
      >
        <span
          className="push-half--left lego-arrow-inline--down"
        />
      </div>
    </div>
  </button>
</div>
`;

exports[`components/Dropdown should not use .oui-arrow-inline--down when icon isEqual to triangle 1`] = `
<div
  className="lego-dropdown-group"
  data-ui-component={true}
  style={
    Object {
      "width": 200,
    }
  }
>
  <button
    className="oui-button"
    disabled={false}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="flex--1 truncate"
      >
        Dropdown
      </div>
      <div
        className="text--right"
      >
        <span
          className="push-half--left oui-arrow-inline--down"
        />
      </div>
    </div>
  </button>
</div>
`;

exports[`components/Dropdown should render children when isOpen is true 1`] = `
<div
  className="lego-dropdown-group"
  data-ui-component={true}
  style={
    Object {
      "width": 200,
    }
  }
>
  <button
    className="oui-button"
    disabled={false}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="flex--1 truncate"
      >
        Dropdown
      </div>
      <div
        className="text--right"
      >
        <span
          className="push-half--left lego-arrow-inline--down"
        />
      </div>
    </div>
  </button>
  <ul>
    <li>
      Manual
    </li>
    <li>
      Maximize Conventions
    </li>
    <li>
      Faster Results
    </li>
  </ul>
</div>
`;

exports[`components/Dropdown should use oui-button--full class when fullWidth is true 1`] = `
<div
  className="lego-dropdown-group"
  data-ui-component={true}
  style={
    Object {
      "width": 200,
    }
  }
>
  <button
    className="oui-button oui-button--full soft--left text--left"
    disabled={false}
    style={
      Object {
        "marginBottom": 2,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="flex--1 truncate"
      >
        Dropdown
      </div>
      <div
        className="text--right"
      >
        <span
          className="push-half--left oui-arrow-inline--down"
        />
      </div>
    </div>
  </button>
</div>
`;
